### ISSUE TITLE
[Backsearch P1] Collect feedback in app

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/land-explorer-front-end#301]
## Description
We wish to collect feedback in the app on how people are using LandExplorer.
People will be able to access the Feedback button at all times, a button on the right hand side.
When users turn on the 'Property Boundaries' layer a pop up appears asking them to fill in the Feedback.
Animations if users say 'No thanks' shrink back to the Button on the RHS to help them know where to find it later.

Based on the research done [here](https://github.com/DigitalCommons/land-explorer-front-end/issues/70), the best first hash solution in terms of storing the data will be to send an API RQ to the backend with the feedback, and to save it in a new DB table. The table needs to include a foreign key to the user that is sending the feedback.

Marcel has created designs for [desktop](https://xd.adobe.com/view/1112be9f-1a2d-4e56-9cb3-d5136938a2e7-4441/screen/bf95af8f-7d4a-4f08-9e9f-136a76f4f0a0) and [mobile](https://xd.adobe.com/view/fcd7a265-2f01-4f67-9a38-d8afa84bdef9-a984/). THere is also additional context in #288 

## Acceptance Criteria
- Button on RHS
- Pop up on Property Boundaries layer, with a 'Dont show again' tickbox. If ticked the user wont see this popup again next time they turn on the PB layer.
- Thank you! to indicate the form submitted.
- Animations that help the user know where to find the button
- Mobile and desktop versions
- Data stored in BE DB linked to the user who is logged in submitting feedback.

**Comment by ms0ur1s** [comment]
Hi @lin-d-hop, everything above has been completed, bar the submission of the form to the BE DB (submission data just logs to the console for the mo), but will hook up with @rogup to sort this out. Also, as well as the pop-up appearing after the user has had the boundary layer switch on for five minutes (an arbitrary number easily changed), it appears when a user attempts to close the app.

### Basic form interactions
https://www.awesomescreenshot.com/video/23889007?key=7ae91c1294747c5e2ffc79fd24e26fe4

### Time delayed pop-up (shortened to 3 seconds for practicality)
https://www.awesomescreenshot.com/video/23889486?key=dbd4f30a812948530bbc9b2dbfbf7372

### Mobile version
https://www.awesomescreenshot.com/video/23889869?key=584516781b08983115437a26aa25ed6c

**Comment by ms0ur1s** [comment]
Hi @lin-d-hop, back end work on this ticket has been completed including but not limited to:
* MySQL seeder and migration files
* Validation, userFeedback model, route and response object.

The front end now successfully submits and populates the user_feedback table which includes the below columns:
* id
* question_1
* question_2
* question_3
* question_4
* user_id (foreign key associated with authenticated user)
* submission date

**Comment by ms0ur1s** [comment]
Hi @lin-d-hop , thanks to @rogup this is now live on staging. 

Rohit also found a couple of idiosyncrasies: 

> The form wasn't fitting in my browser window and the top and bottom were cut off. The blue modal also appeared on top the form when it was already open. And when clicking 'maybe later', the blue modal reappeared after less than a minute.

Thought these would be useful for when you're testing.

**Comment by lin-d-hop** [comment]
Go team! :tada: 

I've just tested myself. Logged into the server to check everything flowed through the to DB correctly. All is looking gooooood :)

I agree with all of the comments. 
My brief notes: 
- on mobile the top and bottom were cut off. 
- The modal reappeared quickly, the reappearing settings could be tweaked. 
- Perhaps the required field warnings could be a more visible colour. 

However what I did not find were any blocking changes. All of these notes are minor things and nothing was unusable or TOO annoying. 

Let's get this live!
